webapp/src/main/java/com/box/l10n/mojito/rest/scheduledjob/ScheduledJobWS.java

```diff
@@ -0,0 +1,129 @@
+
+  static Logger logger = LoggerFactory.getLogger(ScheduledJobWS.class);
+
+  @Autowired ScheduledJobRepository scheduledJobRepository;
+  @Autowired ScheduledJobManager scheduledJobManager;
+
+  @RequestMapping(method = RequestMethod.GET, value = "/api/jobs")
+  @ResponseStatus(HttpStatus.OK)
+  public List<ScheduledJobDTO> getJobs() {
+    List<ScheduledJob> scheduledJobs = scheduledJobRepository.findAll();
+    return scheduledJobs.stream().map(ScheduledJobDTO::new).collect(Collectors.toList());
+  }
+
+  @RequestMapping(method = RequestMethod.GET, value = "/api/jobs/{id}")
+  @ResponseStatus(HttpStatus.OK)
+  public ScheduledJobDTO getJob(@PathVariable String id) {
+    return scheduledJobRepository
+        .findById(id)
+        .map(ScheduledJobDTO::new)
+        .orElseThrow(
+            () ->
+                new ResponseStatusException(HttpStatus.NOT_FOUND, "Job not found with id: " + id));
+  }
+
+  @RequestMapping(method = RequestMethod.POST, value = "/api/jobs/{id}")
+  @ResponseStatus(HttpStatus.OK)
+  public ResponseEntity<ScheduledJobResponse> triggerJob(@PathVariable String id) {
+
+    Optional<ScheduledJob> optionalScheduledJob = scheduledJobRepository.findById(id);
+    if (optionalScheduledJob.isEmpty())
+      return ResponseEntity.status(HttpStatus.NOT_FOUND)
+          .body(new ScheduledJobResponse(ScheduledJobResponse.Status.ERROR, "Job not found"));
+
+    ScheduledJob scheduledJob = optionalScheduledJob.get();
+    JobKey jobKey = scheduledJobManager.getJobKey(scheduledJob);
+
+    try {
+      if (!scheduledJobManager.getScheduler().checkExists(jobKey))
+        return ResponseEntity.status(HttpStatus.NOT_FOUND)
+            .body(new ScheduledJobResponse(ScheduledJobResponse.Status.ERROR, "Job doesn't exist"));
+
+      // Is the job currently running ?
+      // Ignore the trigger request
+      for (JobExecutionContext jobExecutionContext :
+          scheduledJobManager.getScheduler().getCurrentlyExecutingJobs()) {
+        if (jobExecutionContext.getJobDetail().getKey().equals(jobKey)) {
+          return ResponseEntity.status(HttpStatus.CONFLICT)
+              .body(
+                  new ScheduledJobResponse(
+                      ScheduledJobResponse.Status.ERROR,
+                      "Job is currently running, trigger request ignored"));
+        }
+      }
+
+      scheduledJobManager.getScheduler().triggerJob(jobKey);
+      return ResponseEntity.status(HttpStatus.OK)
+          .body(new ScheduledJobResponse(ScheduledJobResponse.Status.SUCCESS, "Job triggered"));
+    } catch (SchedulerException e) {
+      logger.error(
+          "Error triggering job manually, job: {}", jobKey.getName() + ":" + jobKey.getGroup(), e);
+      return ResponseEntity.status(HttpStatus.INTERNAL_SERVER_ERROR)
+          .body(new ScheduledJobResponse(ScheduledJobResponse.Status.ERROR, e.getMessage()));
+    }
+  }
+
+  @RequestMapping(method = RequestMethod.POST, value = "/api/jobs/{id}/toggle")
```

Comment:
Nit: toggle is a strange term for this. I get that you mean it disable and enable a job, but this name feels confusing

Reply:
Honestly I have a preference for separate endpoints for enable and disable because this current endpoint expects the called to know what the previous state was. This could make debugging very difficult.

Example: I called toggled on a job, but I used the wrong job's id. Now the output could either be that it is toggle or not toggled depending on its prior state, which we don't know since this was a mistake. It adds a layer of "what was it before? because that dictates what it is now". Where as the if the endpoint is `/enable` you know if the status code is 200, then it is enabled now. No confusion, no mental gymnastics required

Reply:
Sounds good, I'll change it to two separate endpoints!

Reply:
Done - @byronantak